File: Introductory Problems/Apple Division/appleDivision.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class appleDivision {
    static long ans = Integer.MAX_VALUE;
    private static void helper(List<Integer> wt, int n, int idx, long sum1, long sum2){
        if(idx == n){
            ans = Math.min(ans, Math.abs(sum1-sum2));
            return;
        }
        helper(wt, n, idx+1, sum1+wt.get(idx), sum2);
        helper(wt, n, idx+1, sum1, sum2+wt.get(idx));
        return;
    }
    private static void solve(){
        int n;
        List<Integer> wt = new ArrayList<>();
        Scanner scanner = new Scanner(System.in);
        n = scanner.nextInt();
        for(int i=0;i<n;i++){
            wt.add(scanner.nextInt());
        }
        helper(wt, n, 0, 0, 0);
        System.out.println(ans);
    }
    public static void main(String args[]){
        solve();
    }
}

File: Introductory Problems/Apple Division/apple_division.cpp
#include<bits/stdc++.h>
using namespace std;
#define ll long long int

ll sum1, sum2, ans = INT_MAX;


void helper(vector<int> wt, int idx, int n, ll sum1, ll sum2){
	if(idx == n){
		ans = min(ans, abs(sum1-sum2));
		return;
	}
	helper(wt, idx+1, n, sum1+wt[idx], sum2);
	helper(wt, idx+1, n, sum1, sum2+wt[idx]);
	return;
}

void solve(){
	int n;
	cin>>n;
	vector<int> wt(n);
	for(int i=0;i<n;i++){
		cin>>wt[i];
	}
	helper(wt, 0, n, sum1, sum2);
	cout<<ans;
	return;
	
}

int main(){
	long long testCases;
	testCases = 1;
	// cin>>testCases;
	while(testCases--){
		solve();
	}
	return 0;
}